packages/core-subscriptions/db/subscriptions/helpers.js

```diff
@@ -0,0 +1,304 @@
+      status = SubscriptionStatus.TERMINATED;
+    }
+    return status;
+  },
+  director() {
+    const director = new SubscriptionDirector(this);
+    return director;
+  },
+  setStatus(status, info) {
+    return Subscriptions.updateStatus({
+      subscriptionId: this._id,
+      status,
+      info,
+    });
+  },
+  logs({ limit, offset }) {
+    const selector = { 'meta.subscriptionId': this._id };
+    const logs = Logs.find(selector, {
+      skip: offset,
+      limit,
+      sort: {
+        created: -1,
+      },
+    }).fetch();
+    return logs;
+  },
+  isExpired(referenceDate) {
+    const now = new Date() || referenceDate;
+    const expiryDate = new Date(this.expires);
+    const isExpired = now.getTime() > expiryDate.getTime();
+    return isExpired;
+  },
+});
+
+Subscriptions.createSubscription = async (
+  {
+    productId,
+    quantity,
+    configuration,
+    userId,
+    countryCode,
+    currencyCode,
+    contact,
+    billingAddress,
+    payment,
+    delivery,
+    orderIdForFirstPeriod,
+  },
+  options
+) => {
+  log('Create Subscription', { userId });
+  const subscriptionId = Subscriptions.insert({
+    productId,
+    quantity,
+    configuration,
+    created: new Date(),
+    status: SubscriptionStatus.INITIAL,
+    userId,
+    contact,
+    billingAddress,
+    payment,
+    delivery,
+    currencyCode:
+      currencyCode ||
+      Countries.resolveDefaultCurrencyCode({
+        isoCode: countryCode,
+      }),
+    countryCode,
+  });
+  const subscription = Subscriptions.findOne({ _id: subscriptionId });
+  const period = await subscription.director().nextPeriod();
+  if (period) {
+    await Subscriptions.linkOrderToSubscription({
+      orderId: orderIdForFirstPeriod,
+      subscriptionId,
+      period,
+    });
+  }
+  return subscription.process().sendStatusToCustomer(options);
+};
+
+Subscriptions.linkOrderToSubscription = async ({
+  subscriptionId,
+  period,
+  orderId,
+}) => {
+  const { start, end, isTrial } = period;
+  Subscriptions.update(
+    { _id: subscriptionId },
+    {
+      $push: {
+        periods: {
+          start,
+          end,
+          isTrial,
+          orderId,
+        },
+      },
+      $set: {
+        updated: new Date(),
+      },
+    }
+  );
+  return Subscriptions.findOne({ _id: subscriptionId });
+};
+
+Subscriptions.updateContext = ({ context, subscriptionId }) => {
+  log('Update Arbitrary Context', { subscriptionId });
+  Subscriptions.update(
+    { _id: subscriptionId },
+    {
+      $set: {
+        context,
+        updated: new Date(),
+      },
+    }
+  );
+  return Subscriptions.findOne({ _id: subscriptionId });
+};
+
+Subscriptions.newSubscriptionNumber = () => {
+  let subscriptionNumber = null;
+  const hashids = new Hashids(
+    'unchained',
+    6,
+    'ABCDEFGHIJKLMNOPQRSTUVWXYZ1234567890'
+  );
+  while (!subscriptionNumber) {
+    const randomNumber = Math.floor(Math.random() * (999999999 - 1)) + 1;
```

Comment:
Whats the trick here? Why not just `Math.random() * 1000000000`?

Reply:
I don't know, copy pasted that code from somewhere, maybe so it can never resolve to 0? :)